code/modules/Economy/Wages.dm

```diff
@@ -100,35 +102,36 @@ If all wages are decreased bellow 100%, for example due to the AI spending all t
+			continue
 		if(locate(Acc) in all_station_accounts)
-			if(Acc.wage_gain)
-				adjusted_wage_gain = round((Acc.wage_gain)*payroll_modifier)
-				var/left_from_virtual_wallet = adjusted_wage_gain
-				var/decimal_wage_ratio = 0
-				var/list/obj/item/device/pda/matching_PDAs = list()
-				for(var/obj/item/device/pda/PDA in PDAs)
-					// Only works and does this if ID is in PDA
-					if(PDA?.id?.virtual_wallet)
-						var/datum/pda_app/balance_check/app = locate(/datum/pda_app/balance_check) in PDA.applications
-						if(app && app.linked_db && Acc == app.linked_db.attempt_account_access(PDA.id.associated_account_number, 0, 2, 0))
-							matching_PDAs.Add(PDA)
-				if(matching_PDAs.len)
-					decimal_wage_ratio = Acc.virtual_wallet_wage_ratio/100
-				for(var/obj/item/device/pda/PDA in matching_PDAs)
-					left_from_virtual_wallet -= round(adjusted_wage_gain*(decimal_wage_ratio/matching_PDAs.len))
-					PDA.id.virtual_wallet.money += round(adjusted_wage_gain*(decimal_wage_ratio/matching_PDAs.len))
-					if(round(adjusted_wage_gain*(decimal_wage_ratio/matching_PDAs.len)) > 0)
-						new /datum/transaction(PDA.id.virtual_wallet,"Nanotrasen employee payroll","[round(adjusted_wage_gain*(decimal_wage_ratio/matching_PDAs.len))]",station_account.owner_name)
-				Acc.money += left_from_virtual_wallet
-
-				if(left_from_virtual_wallet > 0)
-					new /datum/transaction(Acc,"Nanotrasen employee payroll","[left_from_virtual_wallet]",station_account.owner_name)
+			adjusted_wage_gain = round((Acc.wage_gain)*payroll_modifier)
+			var/left_from_virtual_wallet = adjusted_wage_gain
+			var/decimal_wage_ratio = 0
+			var/list/obj/item/device/pda/matching_PDAs = list()
+			for(var/obj/item/device/pda/PDA in PDAs)
+				// Only works and does this if ID is in PDA
+				if(PDA?.id?.virtual_wallet)
+					var/datum/pda_app/balance_check/app = locate(/datum/pda_app/balance_check) in PDA.applications
+					if(app && app.linked_db && Acc == app.linked_db.attempt_account_access(PDA.id.associated_account_number, 0, 2, 0))
+						matching_PDAs.Add(PDA)
+			if(matching_PDAs.len)
+				decimal_wage_ratio = Acc.virtual_wallet_wage_ratio/100
+			for(var/obj/item/device/pda/PDA in matching_PDAs)
+				left_from_virtual_wallet -= round(adjusted_wage_gain*(decimal_wage_ratio/matching_PDAs.len))
+				PDA.id.virtual_wallet.money += round(adjusted_wage_gain*(decimal_wage_ratio/matching_PDAs.len))
+				if(round(adjusted_wage_gain*(decimal_wage_ratio/matching_PDAs.len)) > 0)
+					new /datum/transaction(PDA.id.virtual_wallet,"Nanotrasen employee payroll","[round(adjusted_wage_gain*(decimal_wage_ratio/matching_PDAs.len))]",station_account.owner_name)
+			Acc.money += left_from_virtual_wallet
+
+			if(left_from_virtual_wallet > 0)
+				new /datum/transaction(Acc,"Nanotrasen employee payroll","[left_from_virtual_wallet]",station_account.owner_name)
 
 		else 	//non-station accounts get their money from magic, not that these accounts have any wages anyway
 			Acc.money += Acc.wage_gain
```

Comment:
Here was where the station_account managed to get paid infinite magic money out of nowhere.